src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -215,7 +218,12 @@ internal static TypeConverter GetConverter(Type type)
                     bool noTypeConstructor = true;
                     object instance = (TypeConverter)ReflectTypeDescriptionProvider.CreateInstance(converterType, type, ref noTypeConstructor);
                     if (noTypeConstructor)
-                        ReflectTypeDescriptionProvider.IntrinsicTypeConverters[type] = instance;
+                    {
+                        lock (s_syncObject)
```

Comment:
Is this the correct placement, or should it surround the entire get operation?  I put it here under the assumption that it's idempotent and the worst that happens is we end up duplicating some work... let me know if that's not the case.